### DISCUSSION TITLE
Detect the presence of an application (apk

### DISCUSSION THREAD

**Discussion by nathaneltitane** [termux/termux-app#2510] [General]
'am' is currently used in many ways to interact with Android via Termux.
I was wondering if there was some manner of detecting whether or not an apk is present on the system via an 'am' call or how would one go about doing so in a scripted manner.

**Comment by agnostic-apollo** [comment]
`am` is activity manager. `pm` is package manager.

`adb shell pm list packages`

https://developer.android.com/studio/command-line/adb#pm

**Reply by nathaneltitane** [reply]
@agnostic-apollo 

I forgot to mention: within the context of using termux on an android device that serves as a computer with no access to adb...

**Reply by agnostic-apollo** [reply]
Use `adb` from `android-tools` package.

https://www.reddit.com/r/termux/comments/mmu2iu/announce_adb_is_now_packaged_for_termux/

**Reply by nathaneltitane** [reply]
> Use `adb` from `android-tools` package.
> 
> https://www.reddit.com/r/termux/comments/mmu2iu/announce_adb_is_now_packaged_for_termux/

can it be run from the same device of debugging is on?

**Reply by agnostic-apollo** [reply]
Yes, use `localhost` as ip as already mentioned in the post.

**Reply by nathaneltitane** [reply]
> Yes, use `localhost` as ip as already mentioned in the post.

apologies, haven't read up yet. thank you!

**Reply by nathaneltitane** [reply]
![Screenshot_20220105-203211_Termux.jpg](https://user-images.githubusercontent.com/5674757/148313649-779e4eb9-2be7-40b2-bc5b-d57ffcec92c2.jpg)

it errors out...

**Reply by agnostic-apollo** [reply]
That's an ongoing issue with mainly samsung devices.

https://github.com/termux/termux-packages/issues/7946

**Reply by nathaneltitane** [reply]
> That's an ongoing issue with mainly samsung devices.
> 
> https://github.com/termux/termux-packages/issues/7946

well the adb available in termux's repo seems to have an issue.

I got further using rendiix's git I stall script but I still get a connection refused 😑

**Reply by nathaneltitane** [reply]
> > That's an ongoing issue with mainly samsung devices.
> > 
> > https://github.com/termux/termux-packages/issues/7946
> 
> well the adb available in termux's repo seems to have an issue.
> 
> I got further using rendiix's git I stall script but I still get a connection refused 😑

adb tcpip 5555 does not register
adb connect 127.0.0.1:5037 :  connection refused 
adb pair 127.0.0.1:5037 does not return any code for input and authentication...

![Screenshot_20220105-212144_Termux.jpg](https://user-images.githubusercontent.com/5674757/148317908-f1d8c53a-1f99-4ec8-98b4-d200ad853295.jpg)

**Reply by agnostic-apollo** [reply]
Check https://developer.android.com/studio/command-line/adb#connect-to-a-device-over-wi-fi-android-11+

pair and connect ip and ports are different.

**Reply by nathaneltitane** [reply]
> > > That's an ongoing issue with mainly samsung devices.
> > > 
> > > https://github.com/termux/termux-packages/issues/7946
> > 
> > well the adb available in termux's repo seems to have an issue.
> > 
> > I got further using rendiix's git I stall script but I still get a connection refused 😑
> 
> adb tcpip 5555 does not register
> adb connect 127.0.0.1:5037 :  connection refused 
> adb pair 127.0.0.1:5037 does not return any code for input and authentication...
> 
> ![Screenshot_20220105-212144_Termux.jpg](https://user-images.githubusercontent.com/5674757/148317908-f1d8c53a-1f99-4ec8-98b4-d200ad853295.jpg)


gone through that using the supplied code in debug tools, still no dice...

added some debug in ticket
